libphx/src/window.rs

```diff
@@ -27,165 +29,184 @@ impl Drop for Window {
     }
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn Window_Create(
-    title: *const libc::c_char,
-    x: i32,
-    y: i32,
-    sx: i32,
-    sy: i32,
-    mode: WindowMode,
-) -> Box<Window> {
-    let modeComplete = mode | SDL_WindowFlags::SDL_WINDOW_OPENGL as WindowMode;
-    // | SDL_WindowFlags::SDL_WINDOW_ALLOW_HIGHDPI as WindowMode;
-    let handle = SDL_CreateWindow(title, x, y, sx, sy, modeComplete);
-    let context = SDL_GL_CreateContext(handle);
-    if context.is_null() {
-        CFatal!("Failed to create OpenGL context for window");
-    }
-    OpenGL_Init();
-
-    Box::new(Window {
-        handle,
-        context,
-        mode: modeComplete,
-        cursor: std::ptr::null_mut(),
-    })
-}
+#[luajit_ffi_gen::luajit_ffi(meta = true, managed = true)]
```

Comment:
`managed` basically makes sure to generate the `managed()` and `free()` metamethods, which we don't want in the case that we're binding something like the `Bit` package right?

I'm curious, when would we have `meta = true` but `managed = false`?

Reply:
I think I've seen an example of that but I'll have to check it again

Reply:
Found example. Quat.lua
But if it's just a mistake I can add managed to all metatype sections

Reply:
Lets leave it for now, I think it's nice to have that control. Perhaps once we've converted all of the types, we might notice some patterns to automate / remove this parameter.